forgerock-authenticator/src/main/java/org/forgerock/android/authenticator/Account.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.authenticator;
+
+import org.forgerock.android.authenticator.util.SortedList;
+
+import java.util.List;
+
+/**
+ * Account model represents an identity for the user with an issuer. It is possible for a user to
+ * have multiple accounts provided by a single issuer, however it is not possible to have multiple
+ * accounts from with same issuer for the same account name.
+ */
+public class Account extends ModelObject<Account> {
+
+    /** Unique identifier of Account */
+    private final String id;
+    /** Issuer of the account */
+    private final String issuer;
+    /** AccountName, or Username of the account for the issuer */
+    private final String accountName;
+    /** URL of Account's logo image */
+    private final String imageURL;
+    /** HEX Color code in String for Account */
+    private final String backgroundColor;
+    /** An array of Mechanism associated with current Account */
+    private final List<Mechanism> mechanisms;
```

Comment:
This is not used, I like the idea of creating relationship between data model, however, it seem the StorageClient interface did not utilize it. The interface defines store and retrieves independently, they will eventually have data integrity issue, for example, we remove all account but not mechanism, or we remove Push Mechanism but did not remove Notification?

Reply:
This is just a placeholder, I was thinking about a Lazy initialization for those objects. I believe iOS implemented that way, but it is not decided yet. We will review it and consider this feedback.